x-pack/plugins/lens/public/shared_components/legend_settings_popover.tsx

```diff
@@ -158,6 +212,63 @@ export const LegendSettingsPopover: React.FunctionComponent<LegendSettingsPopove
         position={position}
         onPositionChange={onPositionChange}
       />
+      <EuiFormRow
+        display="columnCompressedSwitch"
+        label={i18n.translate('xpack.lens.shared.truncateLegend', {
+          defaultMessage: 'Truncate legend text',
+        })}
+      >
+        <TooltipWrapper
+          tooltipContent={i18n.translate('xpack.lens.shared.legendVisibleTooltip', {
+            defaultMessage: 'Requires legend to be shown',
+          })}
+          condition={mode === 'hide'}
+          position="top"
+          delay="regular"
+          display="block"
+        >
+          <EuiSwitch
+            compressed
+            label={i18n.translate('xpack.lens.shared.truncateLegend', {
+              defaultMessage: 'Truncate legend text',
+            })}
+            data-test-subj="lens-legend-truncate-switch"
+            showLabel={false}
+            disabled={mode === 'hide'}
+            checked={truncate ?? true}
+            onChange={onTruncateLegendChange}
+          />
+        </TooltipWrapper>
+      </EuiFormRow>
+      <EuiFormRow
+        label={i18n.translate('xpack.lens.shared.maxLinesLabel', {
+          defaultMessage: 'Maximum legend lines',
```

Comment:
Similar to the previous comment, consider changing to the following:

```suggestion
          defaultMessage: 'Maximum lines',
```